crates/gossip/proto/gossip.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package kitsune2.gossip;
+
+// A Kitsune2 gossip protocol message.
+message K2GossipProto {
+  oneof gossip_message {
```

Comment:
oneof has some fiddly [backward compatibility issues](https://protobuf.dev/programming-guides/proto3/#backward).

Another way of handling this would be to have an `enum` that types the message, then a further encoded bytes field to be parsed based on the enum. This would mean more complexity in our code, though, so I'm okay with the fiddly-ness so long as we're aware of it.

Reply:
Hmm yeah okay. So I guess if we're careful with an enum then we'd be a bit more able to implement backwards compatibility ourselves but aren't we going to have similar problems?

I'd suggest that if we really want to make major changes then we can test them against an older Kitsune2. If we have to make breaking changes then we define a new `oneof` field and have the code behave differently depending on which one is present until we can deprecate the old field.

Reply:
Yep, I'm fine with that approach.